### PR TITLE
Version 0.4.0

### PR DESCRIPTION
**Pull request by Alexandre-saddour**
This PR introduce the following changes

New Usecases
- `CallContractUsecase` -> equivalent to `erdpy contract call`
- `QueryContractHexUsecase` -> equivalent to `vm-values/hex` 
- `QueryContractStringUsecase` -> equivalent to `vm-values/string`
- `QueryContractIntUsecase` -> equivalent to `vm-values/int`

Update & clean old data
- `returnData` from `vm-values/query` is now decoded and accessible in several types (asBase64,asString,asHex,asBigInt)
- rename `QuerySmartContractUsecase` to `QueryContractUsecase` (+ rename internal input fields)
- add missing fields to `TransactionOnNetwork.ScResult`

Other
- add several tests (+ clean test packages).
- clean pom file.

### CODE REVIEW

File: src/main/java/com/elrond/erdkotlin/data/vm/VmRepositoryImpl.kt [hunk 1]
```diff
@@ -2,14 +2,28 @@ package com.elrond.erdkotlin.data.vm
...
 
 internal class VmRepositoryImpl(private val elrondProxy: ElrondProxy) : VmRepository {
 
-    override fun queryContract(smartContractQuery: SmartContractQuery): SmartContractOutput {
-        return requireNotNull(elrondProxy.queryContract(smartContractQuery).data).data.toDomain()
+    override fun queryContract(queryContractInput: QueryContractInput): QueryContractOutput {
+        return requireNotNull(elrondProxy.queryContract(queryContractInput).data).data.toDomain()
+    }
+
+    override fun queryContractHex(queryContractInput: QueryContractInput): QueryContractStringOutput {
```
Comment (bogdan-rosianu) [inline src/main/java/com/elrond/erdkotlin/data/vm/VmRepositoryImpl.kt:17]: Why wouldn't extend the `queryContract` function and specify within a parameter the way to interpret the result? A call would look similar to `queryContract(input, ContractQueryEnum.Hex)`  and do the query to the endpoint specific to the enum value (as well as the return object formatting)

Comment (Alexandre-saddour) [reply src/main/java/com/elrond/erdkotlin/data/vm/VmRepositoryImpl.kt:17]: It sounds good for the input part.

However, on the output side, I do not see how I could make it work.
The issue is the return type.
- `queryContract ` returns a complex object (containing may fields)
- `queryContractHex` and `queryContractString` both return a single `String`
- `queryContractInt` returns a single `BigInteger`

The only thing I can do is merging `queryContractHex`, `queryContractString` and `queryContractInt` in the repository (which is in the private part of the SDK) by returning a String.
Then I can format that string in the usecases
But I will have to keep separate usecases to differentiate the `String` and `BigInteger` cases.

I cannot merge `queryContract` with its siblings at all.

File: src/main/java/com/elrond/erdkotlin/domain/dns/GetDnsRegistrationCostUsecase.kt [hunk 2]
```diff
@@ -17,8 +16,8 @@ class GetDnsRegistrationCostUsecase internal constructor(
     fun execute(dnsAddress: Address): BigInteger {
         val result = queryContractUsecase.execute(dnsAddress, "getRegistrationCost")
         return when {
-            result.returnData.isNullOrEmpty() || result.returnData[0].isEmpty() -> 0.toBigInteger()
-            else -> BigInteger(Base64.decode(result.returnData[0]))
+            result.returnData.isNullOrEmpty() -> 0.toBigInteger()
```
Comment (ditzdragos) [inline src/main/java/com/elrond/erdkotlin/domain/dns/GetDnsRegistrationCostUsecase.kt:19]: Let's use BigInteger.ZERO

File: src/main/java/com/elrond/erdkotlin/domain/sc/CallContractUsecase.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    fun execute(
+        account: Account,
+        wallet: Wallet,
+        networkConfig: NetworkConfig,
+        gasPrice: Long,
+        gasLimit: Long,
+        contractAddress: Address,
+        funcName: String,
+        args: List<String> = emptyList(),
+        value: BigInteger = 0.toBigInteger(),
```
Comment (ditzdragos) [inline src/main/java/com/elrond/erdkotlin/domain/sc/CallContractUsecase.kt:29]: Let's use BigInteger.ZERO

Comment (Alexandre-saddour) [reply src/main/java/com/elrond/erdkotlin/domain/sc/CallContractUsecase.kt]: Done

File: src/main/java/com/elrond/erdkotlin/domain/dns/GetDnsRegistrationCostUsecase.kt (hunk 2 above)
Comment (Alexandre-saddour) [reply src/main/java/com/elrond/erdkotlin/domain/dns/GetDnsRegistrationCostUsecase.kt]: Done

**Review by bogdan-rosianu — APPROVED** [review]

File: src/main/java/com/elrond/erdkotlin/data/vm/VmRepositoryImpl.kt (hunk 1 above)
Comment (bogdan-rosianu) [reply src/main/java/com/elrond/erdkotlin/data/vm/VmRepositoryImpl.kt:17]: that's right. :+1:

**Review by ditzdragos — APPROVED** [review]

**Comment by Alexandre-saddour** [comment]
Hello, can we merge this week ?

**Review by miiu96 — APPROVED** [review]

**Pull request merged**